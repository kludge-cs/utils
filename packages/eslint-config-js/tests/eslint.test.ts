// @ts-ignore: using ts-jest for different directory creates false warning
import eslintConfig from "../src";

describe("ESLint Config", () => {
	test("should export rules", () => {
		expect(eslintConfig.root).toBe(true);
		expect(eslintConfig).toMatchSnapshot();
	});

	test("should pass for require", () => {
		expect(require("../src")).toStrictEqual(eslintConfig);
	});
});
