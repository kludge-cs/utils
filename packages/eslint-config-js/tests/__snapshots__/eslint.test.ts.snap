// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Config should export rules 1`] = `
Object {
  "env": Object {
    "commonjs": true,
    "es2021": true,
    "es6": true,
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "plugin:node/recommended",
  ],
  "parserOptions": Object {
    "ecmaVersion": 6,
    "extraFileExtensions": Array [
      ".mjs",
    ],
    "sourceType": "module",
  },
  "root": true,
  "rules": Object {
    "accessor-pairs": "error",
    "array-bracket-newline": "off",
    "array-bracket-spacing": Array [
      "error",
      "never",
    ],
    "array-callback-return": Array [
      "error",
      Object {
        "allowImplicit": true,
        "checkForEach": true,
      },
    ],
    "array-element-newline": "off",
    "arrow-body-style": "off",
    "arrow-parens": Array [
      "error",
      "always",
    ],
    "arrow-spacing": Array [
      "error",
      Object {
        "after": true,
        "before": true,
      },
    ],
    "block-scoped-var": "error",
    "block-spacing": "error",
    "brace-style": Array [
      "error",
      "1tbs",
    ],
    "camelcase": "error",
    "capitalized-comments": "off",
    "comma-dangle": "error",
    "comma-spacing": Array [
      "error",
      Object {
        "after": true,
        "before": false,
      },
    ],
    "comma-style": Array [
      "error",
      "last",
    ],
    "complexity": "error",
    "computed-property-spacing": "error",
    "consistent-return": "off",
    "consistent-this": "error",
    "curly": "error",
    "default-case": "error",
    "default-case-last": "error",
    "default-param-last": "error",
    "dot-location": Array [
      "error",
      "property",
    ],
    "dot-notation": Array [
      "error",
      Object {
        "allowKeywords": true,
      },
    ],
    "eol-last": Array [
      "error",
      "always",
    ],
    "eqeqeq": "error",
    "for-direction": "error",
    "func-call-spacing": "error",
    "func-name-matching": "error",
    "func-names": "error",
    "func-style": Array [
      "error",
      "declaration",
    ],
    "function-call-argument-newline": Array [
      "error",
      "consistent",
    ],
    "function-paren-newline": "off",
    "generator-star-spacing": "error",
    "getter-return": "error",
    "grouped-accessor-pairs": "error",
    "guard-for-in": "warn",
    "id-length": "error",
    "id-match": "error",
    "implicit-arrow-linebreak": "error",
    "indent": Array [
      "error",
      "tab",
      Object {
        "SwitchCase": 1,
      },
    ],
    "init-declarations": "off",
    "jsx-quotes": "error",
    "key-spacing": "error",
    "keyword-spacing": Array [
      "error",
      Object {
        "after": true,
        "before": true,
      },
    ],
    "line-comment-position": "warn",
    "linebreak-style": Array [
      "error",
      "unix",
    ],
    "lines-around-comment": "error",
    "lines-between-class-members": Array [
      "error",
      "always",
      Object {
        "exceptAfterSingleLine": true,
      },
    ],
    "max-classes-per-file": "error",
    "max-depth": "error",
    "max-len": Array [
      "error",
      Object {
        "ignoreComments": true,
        "ignoreTemplateLiterals": true,
      },
    ],
    "max-lines": "off",
    "max-lines-per-function": "error",
    "max-nested-callbacks": "error",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "error",
    "multiline-comment-style": Array [
      "error",
      "bare-block",
    ],
    "multiline-ternary": Array [
      "error",
      "always-multiline",
    ],
    "new-cap": "error",
    "new-parens": "error",
    "newline-per-chained-call": "error",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "error",
    "no-caller": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "error",
    "no-console": "off",
    "no-constant-condition": Array [
      "error",
      Object {
        "checkLoops": false,
      },
    ],
    "no-constructor-return": "error",
    "no-continue": "error",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-div-regex": "warn",
    "no-duplicate-imports": "error",
    "no-else-return": "error",
    "no-empty-function": "error",
    "no-eval": "off",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-extra-parens": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": "error",
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-inline-comments": "warn",
    "no-invalid-this": "error",
    "no-irregular-whitespace": "off",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-loss-of-precision": "error",
    "no-magic-numbers": "off",
    "no-mixed-operators": "error",
    "no-multi-assign": "error",
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-multiple-empty-lines": "error",
    "no-negated-condition": "off",
    "no-nested-ternary": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-plusplus": "error",
    "no-promise-executor-return": "error",
    "no-proto": "error",
    "no-restricted-exports": "error",
    "no-restricted-globals": "error",
    "no-restricted-imports": "error",
    "no-restricted-properties": "error",
    "no-restricted-syntax": "error",
    "no-return-assign": "error",
    "no-return-await": "off",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow": "error",
    "no-tabs": Array [
      "error",
      Object {
        "allowIndentationTabs": true,
      },
    ],
    "no-template-curly-in-string": "warn",
    "no-ternary": "off",
    "no-throw-literal": "error",
    "no-trailing-spaces": "error",
    "no-undef-init": "error",
    "no-undefined": "off",
    "no-underscore-dangle": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": Array [
      "error",
      Object {
        "defaultAssignment": true,
      },
    ],
    "no-unreachable-loop": "error",
    "no-unsafe-negation": Array [
      "error",
      Object {
        "enforceForOrderingRelations": true,
      },
    ],
    "no-unused-expressions": "error",
    "no-unused-vars": "error",
    "no-use-before-define": "error",
    "no-useless-backreference": "error",
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": Array [
      "error",
      Object {
        "allowAsStatement": true,
      },
    ],
    "no-warning-comments": "warn",
    "no-whitespace-before-property": "error",
    "node/no-missing-import": Array [
      "error",
      Object {
        "tryExtensions": Array [
          ".js",
          ".json",
          "./index.js",
        ],
      },
    ],
    "node/no-missing-require": Array [
      "error",
      Object {
        "tryExtensions": Array [
          ".js",
          ".json",
          "./index.js",
        ],
      },
    ],
    "nonblock-statement-body-position": "error",
    "object-curly-newline": "error",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "object-shorthand": Array [
      "error",
      "properties",
      Object {
        "avoidQuotes": true,
      },
    ],
    "one-var": "off",
    "one-var-declaration-per-line": "error",
    "operator-assignment": "error",
    "operator-linebreak": Array [
      "error",
      "before",
    ],
    "padded-blocks": Array [
      "error",
      "never",
    ],
    "padding-line-between-statements": "error",
    "prefer-arrow-callback": "error",
    "prefer-const": "error",
    "prefer-destructuring": "off",
    "prefer-exponentiation-operator": "error",
    "prefer-named-capture-group": "off",
    "prefer-numeric-literals": "error",
    "prefer-object-spread": "error",
    "prefer-promise-reject-errors": "error",
    "prefer-regex-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "error",
    "quote-props": "off",
    "quotes": Array [
      "error",
      "double",
    ],
    "radix": "error",
    "require-atomic-updates": "error",
    "require-await": "off",
    "require-unicode-regexp": "off",
    "rest-spread-spacing": "error",
    "semi": "error",
    "semi-spacing": "error",
    "semi-style": Array [
      "error",
      "last",
    ],
    "sort-imports": "error",
    "sort-keys": "off",
    "sort-vars": "error",
    "space-before-blocks": "error",
    "space-before-function-paren": "off",
    "space-in-parens": Array [
      "error",
      "never",
    ],
    "space-infix-ops": "error",
    "space-unary-ops": "error",
    "spaced-comment": Array [
      "error",
      "always",
    ],
    "switch-colon-spacing": "error",
    "symbol-description": "error",
    "template-curly-spacing": Array [
      "error",
      "never",
    ],
    "template-tag-spacing": "error",
    "unicode-bom": Array [
      "error",
      "never",
    ],
    "vars-on-top": "error",
    "wrap-iife": "error",
    "wrap-regex": "off",
    "yield-star-spacing": "error",
    "yoda": Array [
      "error",
      "never",
    ],
  },
}
`;
